base.go

```diff
@@ -348,47 +348,51 @@ func getBuilt(o BaseLayerOpts, sfm StackerFiles) error {
 	// may not have been copied before and the final `umoci repack` expects
 	// them to be there.
 	targetName := o.Name
-	baseLayerDefinition, ok := sfm.LookupLayerDefinition(o.Layer.From.Tag)
-	if !ok {
-		return errors.Errorf("missing base layer: %s?", o.Layer.From.Tag)
-	}
-
-	if baseLayerDefinition.BuildOnly {
-		baseInputTag, err := baseLayerDefinition.From.ParseTag()
-		if err != nil {
-			return err
+	base := o.Layer
+	for {
+		// Iterate through base layers until we find the first one which is not BuiltType or BuildOnly
+		// Need to declare ok separately, if we do it in the same line as
+		// assigning the new value to base, base would be a new variable only in the scope
+		// of this iteration and we never meet the condition to exit the loop
+		var ok bool
+		base, ok = sfm.LookupLayerDefinition(base.From.Tag)
+		if !ok {
+			return fmt.Errorf("missing base layer: %s?", base.From.Tag)
 		}
 
-		// Ok, we didn't generate an OCI diff for whatever mutated this
-		// layer. So either 1. the layer was sourced
-		// from something and is in the OCI cache, so we can copy at
-		// least the base's base over to make the delta smaller or 2.
-		// it was sourced from e.g. a tar file, in which case there's
-		// nothing we can do besides initialize an empty oci tag and
-		// generate the whole thing.
-		cacheDir := path.Join(o.Config.StackerDir, "layer-bases", "oci")
-		err = lib.ImageCopy(lib.ImageCopyOpts{
-			Src:  fmt.Sprintf("oci:%s:%s", cacheDir, baseInputTag),
-			Dest: fmt.Sprintf("oci:%s:%s", o.Config.OCIDir, targetName),
-		})
-		if err != nil {
-			// give up and just create an empty image
-			return umociInit(o)
+		if base.From.Type != BuiltType || !base.BuildOnly {
+			break
 		}
+	}
+
+	// Nothing to do here -- we didn't import any base layers.
+	if base.From.Type != DockerType && base.From.Type != OCIType && base.From.Type != ZotType && base.From.Type != BuiltType {
```

Comment:
I think this is still wrong, we always have to do *something* here for the squashfs case, because the resulting code updates a tag it expects to exist.